lib/ClangImporter/ImportDecl.cpp

```diff
@@ -6298,6 +6298,29 @@ void SwiftDeclConverter::recordObjCOverride(AbstractFunctionDecl *decl) {
   }
 }
 
+static bool areIndicesEqual(SubscriptDecl *subscript1,
+                            SubscriptDecl *subscript2) {
+  auto &params1 = *subscript1->getIndices();
+  auto &params2 = *subscript2->getIndices();
+
+  if (params1.size() != params2.size())
+    return false;
+
+  for (unsigned i : indices(params1)) {
+    if (!params1[i]->getInterfaceType()->isEqual(
+          params2[i]->getInterfaceType())) {
+      return false;
+    }
+
+    if (params1[i]->getValueOwnership() !=
+        params2[i]->getValueOwnership()) {
+      return false;
+    }
+  }
```

Comment:
I feel like this helper doesn't belong here. Checking that two ParameterLists match exactly seems like a generally useful feature.

(Also, `const` please.)

Reply:
I'll rename it to areParameterTypesEqual() and make it take const ParameterLists instead of SubscriptDecl, but I don't want to move it to a more general place just yet because it also has the specific behavior that it ignores labels. Let's revisit this later if we need it again.

Reply:
I think the ignoring labels part makes sense for multiple reasons: labels aren't part of the type, and we already have full names. But taking ParameterLists is really the important bit, so I guess it's okay if it stays here.